### PR TITLE
chore: enable strict ordering of transactions

### PR DESCRIPTION
**Pull request by MatusKysel**
## Purpose
Fixes: #1516

### CODE REVIEW

**Review by JakubFornadel — APPROVED** [review]

File: tests/util_test/util.hpp [hunk 1]
```diff
@@ -124,6 +124,9 @@ inline auto make_node_cfgs(uint count) {
       cfg.chain.final_chain.state.genesis_balances[root_node_addr] = 9007199254740991;
       auto& dpos = *cfg.chain.final_chain.state.dpos;
       dpos.genesis_state[root_node_addr][root_node_addr] = dpos.eligibility_balance_threshold;
+      // As test are badly written let's disable it for now
+      cfg.chain.final_chain.state.execution_options.disable_nonce_check = true;
```
Comment (gaoqi8649) [inline tests/util_test/util.hpp:128]: This will cause lots of unit tests disable nonce check. Which unit tests will test nonce checking?

Comment (MatusKysel) [reply tests/util_test/util.hpp:128]: With this change we have 2 levels of nonce checking, first one when we verifying transaction and second one when we execute transaction. This flag is disabling second one, that is part of EVM and not this PR. Anyway I am not disabling anything new, just moving this flag only for tests. If we want to enable it by default, we need to rework a lot of tests and that's for separate task

Comment (gaoqi8649) [reply tests/util_test/util.hpp:128]: I see. Thanks!

**Pull request merged**